File: src/main/java/org/goafabric/catalog/job/chargeitem/ChargeItemEo.java
package org.goafabric.catalog.job.chargeitem;

import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Version;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.relational.core.mapping.Table;

import java.util.UUID;

@Table(name = "chargeitem", schema = "catalog")
@Document("chargeitem")
public record ChargeItemEo(
    @Id String id,
    @Version Long version,
    String code,
    String display,
    Double price
) {
    public ChargeItemEo(String code, String display, Double price) {
        this(UUID.randomUUID().toString(), null, code, display, price);
    }
}

File: src/main/kotlin/org/goafabric/catalog/job/chargeitem/ChargeItemEo.kt
package org.goafabric.catalog.job.chargeitem

import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Version
import org.springframework.data.relational.core.mapping.Table
import java.util.*

@Table(name = "chargeitem", schema = "catalog") //@Document("chargeitem")
data class ChargeItemEo(
    @Id val id: String = UUID.randomUUID().toString(),
    @Version val version: Long? = null,
    val code: String?,
    val display: String?,
    val price: Double?
)